hw/ip/prim/util/gen-lfsr-seed.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+r"""This script generates random seeds and state permutations for LFSRs
+and outputs them them as a packed SV logic vectors suitable for use with
+prim_lfsr.sv.
+"""
+import argparse
+import random
+import textwrap
+import logging as log
+from math import ceil, log2
+
+
+SV_INSTRUCTIONS = """
+------------------------------------------------
+| COPY PASTE THE CODE TEMPLATE BELOW INTO YOUR |
+| RTL CODE, INLUDING THE COMMENT IN ORDER TO   |
+| EASE AUDITABILITY AND REPRODUCIBILITY.       |
+------------------------------------------------
+"""
+
+
+def _get_random_data_hex_literal(width):
+    """ Fetch 'width' random bits and return them as hex literal"""
+    width = int(width)
+    literal_str = hex(random.getrandbits(width))
+    literal_str = str(width) + "'h" + literal_str[2:]
+    return literal_str
+
+
+def _get_random_perm_hex_literal(numel):
+    """ Compute a random permutation of 'numel' elements and
+    return as packed hex literal"""
+    num_elements = int(numel)
+    width = int(ceil(log2(num_elements)))
+    idx = [x for x in range(num_elements)]
+    random.shuffle(idx)
+    literal_str = ""
+    for k in idx:
+        literal_str += format(k, '0' + str(width) + 'b')
+    # convert to hex for space efficiency
+    literal_str = hex(int(literal_str, 2))
+    literal_str = str(width * numel) + "'h" + literal_str[2:]
+    return literal_str
+
+
+def _wrapped_docstring():
+    '''Return a text-wrapped version of the module docstring'''
+    paras = []
+    para = []
+    for line in __doc__.strip().split('\n'):
+        line = line.strip()
+        if not line:
+            if para:
+                paras.append('\n'.join(para))
+                para = []
+        else:
+            para.append(line)
+    if para:
+        paras.append('\n'.join(para))
+
+    return '\n\n'.join(textwrap.fill(p) for p in paras)
+
+
+def main():
+    log.basicConfig(level=log.INFO,
+                    format="%(asctime)s - %(message)s",
+                    datefmt="%Y-%m-%d %H:%M")
+
+    parser = argparse.ArgumentParser(
+        prog="gen-lfsre-perm",
+        description=_wrapped_docstring(),
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('-w',
```

Comment:
I would add an optional name argument to that is added to the parameter and type names in the SystemVerilog template, e.g., `lfsr_seed_t` would become `prng_lfsr_seed_t` or `clr_lfsr_seed_t` and `RndCnstLfsrSeedDefault` would become `RndCnstPrngLfsrSeedDefault`. This is useful if an IP has with multiple LFSRs of potentially different sizes such as AES.

Reply:
Ok, I've added a `--prefix` switch to do that:
```
$ hw/ip/prim/util/gen-lfsr-seed.py --prefix FooBar

------------------------------------------------
| COPY PASTE THE CODE TEMPLATE BELOW INTO YOUR |
| RTL CODE, INLUDING THE COMMENT IN ORDER TO   |
| EASE AUDITABILITY AND REPRODUCIBILITY.       |
------------------------------------------------


// These LFSR parameters have been generated with
// $ hw/ip/prim/util/gen-lfsr-seed.py --width 32 --seed 913503156 --prefix "FooBar"
localparam int FooBarLfsrWidth = 32;
typedef logic [FooBarLfsrWidth-1:0] foo_bar_lfsr_seed_t;
typedef logic [FooBarLfsrWidth-1:0][$clog2(FooBarLfsrWidth)-1:0] foo_bar_lfsr_perm_t;
localparam lfsr_seed_t RndCnstFooBarLfsrSeedDefault = 32'h45fe6df7;
localparam lfsr_perm_t RndCnstFooBarLfsrPermDefault =
    160'h1f1910a8fb74f4f049b5275382aed0a67df122e6;
```

Reply:
Awesome, thanks!